addons/crm_mail_plugin/tests/test_crm_mail_plugin.py

```diff
@@ -54,3 +39,82 @@ def test_get_contact_data(self):
             msg="The first lead belongs to the first partner, it should be returned")
         self.assertFalse([lead for lead in result["leads"] if lead["lead_id"] == lead_2.id],
             msg="The second lead does not belong to the first partner, it should not be returned")
+
+    @mock_auth_method_outlook('employee')
+    def test_crm_lead_create_multi_company(self):
+        """ Test that creating a record using the mail plugin for a contact belonging to a different company than the
+            default company of the user does not result in any issues.
+        """
+        # create Company_A and Company_B
```

Comment:
to much comment is not good :) (here it's explicit enough)
```suggestion
```

only this one in this test is relevant imo
```
# Check that the created lead record has the correct company and return the lead_id
```

Reply:
@std-odoo Thanks for your review.
I modified it. But I kept the comments that are directly related to the test.